// Copyright 2018 Google Cloud Platform Proxy Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.


syntax = "proto3";

package google.api_proxy.agent;

option cc_generic_services = true;

import "google/protobuf/duration.proto";

// API Proxy Agent Service
service AgentService {
  // Returns an access token with its expiration time.
  rpc GetAccessToken(GetAccessTokenRequest) returns (GetAccessTokenResponse) {}
}

message GetAccessTokenRequest {
}

message GetAccessTokenResponse {
  // The access token fetched from API Proxy Agent.
  string access_token = 1;

  // Duration indicating how long the token will expire.
  google.protobuf.Duration expires_in = 4;
}
